### PR TITLE
impl: Kyle delete cards and iterate backwards

### PR DESCRIPTION
**Pull request by emich-kyle**
Sorry, I've accidentally merged my "previous card" branch into the "delete card" branch so I will close two PBIs in this pull request this time:

There is a previous card button that allows the user to iterate backwards through cards.

Closes issue #57 

Delete cards requires refactoring of the flashdeck constant from reference to state so that the viewer can rerender when cards are deleted. The user can delete any card whether it is on tile view or single card view.

Deleted card changes appear both on the .json when you download, and dynamically on the viewer as the user deletes them. The delete button only appears when you are in edit view.

Closes issue #50

### CODE REVIEW

File: frontend/flashfolio/src/LoginButton.js [hunk 1]
```diff
@@ -8,13 +8,12 @@ import { loginContext } from "./App.js";
...
 
 
-	const { loginState, setLoginState } = useContext(loginContext);
+	const { setLoginState } = useContext(loginContext);
 
 	const success = async (res) => {
 		console.log('Logged in');
 		setLoginState(res);
-		await notifyUserLogin(res.tokenId, res.profileObj);
-		localStorage.setItem("userCache", JSON.stringify(res));
```
Comment (rrstem) [inline frontend/flashfolio/src/LoginButton.js:17]: Why did you remove this?

File: frontend/flashfolio/src/Calls.js [hunk 2]
```diff
@@ -86,25 +85,24 @@ export async function cloneDeck(token, deck) {
 	console.log(resp)
 	return resp.json();
 }
+
 export async function notifyUserLogin(token, profileObj) {
 	let reqOpt = {
 		method: "POST",
-		headers: {"Content-Type": "application/json"},
-		body: JSON.stringify({"Token": token, "NickName": profileObj.name, "ProfilePicture":profileObj.imageUrl}),
```
Comment (rrstem) [inline frontend/flashfolio/src/Calls.js:93]: Why did you remove this?

File: backend/flashfolio/user.go [hunk 3]
```diff
@@ -87,9 +64,7 @@ func UserLoginReq(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var req struct {
-		Token          string `json:"Token"`
-		NickName       string `json:"NickName"`
-		ProfilePicture string `json:"ProfilePicture"`
```
Comment (rrstem) [inline backend/flashfolio/user.go:92]: Why did you remove this?

File: backend/flashfolio/main.go [hunk 4]
```diff
@@ -150,35 +150,13 @@ func saveDeckReq(w http.ResponseWriter, r *http.Request) {
...
-	err = collection.FindOne(ctx, bson.D{{Key: "id", Value: req.Deck.ID}}).Decode(&realDeck)
-	if err != nil {
-		/* That deck doesn't exist => 404 */
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	if tokenInfo.UserId != realDeck.Owner {
-		/* Not the deck owner => Forbidden */
-		w.WriteHeader(http.StatusForbidden)
-		return
```
Comment (rrstem) [inline backend/flashfolio/main.go:180]: Again, why are you removing all this stuff?

File: backend/flashfolio/user.go [hunk 5]
```diff
@@ -13,21 +13,13 @@ import (
...
 
 type User struct {
-	ID             string `json:"ID" bson:"id"`
-	Email          string `json:"Email" bson:"email"`
-
-	// Google User name
-	NickName       string `json:"NickName" bson:"nickname"`
-
-	// Google Profile Picture
-	ProfilePicture string `json:"ProfilePicture" bson:"profilepicture"`
```
Comment (rrstem) [inline backend/flashfolio/user.go:27]: ProfilePicture, NickName -- why did you remove these.

File: backend/flashfolio/user.go [hunk 6]
```diff
@@ -58,21 +50,6 @@ func GetUserByID(id string, ctx context.Context) (*User, error) {
...
-
-	// Create new entry if one does not exist
-	opt := options.Replace().SetUpsert(create)
-
-	// set up filter to locate document with identical user generated ID
-	filter := bson.D{{Key: "id", Value: user.ID}}
-
-	// Replace document within mongo if found.
-	collection.ReplaceOne(ctx, filter, user, opt)
-}
```
Comment (rrstem) [inline backend/flashfolio/user.go:74]: Why did you just get rid of this entire method?????

**Review by rrstem — CHANGES_REQUESTED** [review]
You seem to have removed a lot of other peoples work when resolving these conflicts -- e.g. deck privacy, user profile pictures. As such I cannot approve this.

**Comment by emich-kyle** [comment]
Sorry about that i did not even know that happened. my Git app must have not synced properly and it kept old changes. I'll revert and try again
